### ISSUE TITLE
[Bug][ROCm] GLM-5.3-MXFP4 TP8 + EAGLE GPU memory fault under high-concurrency long-context load

### ISSUE THREAD

**Issue by jhinpan** [sgl-project/sglang#37648]
### Checklist

- [x] I searched related issues and found a similar older report, #23784, but no documented fix or linked resolution.
- [ ] I have not yet reproduced this on current `main`; the deployed image is from 2026-08-29. I will follow up with current-main and ablation results.
- [x] I understand that a minimal reproducer is needed for root-cause work. This report records the complete failure evidence now; a smaller reproducer is follow-up work.
- [x] This is a bug report rather than a general usage question.
- [x] This report is in English.

### Describe the bug

An 8x MI355X SGLang server running `OneNexus/GLM-5.3-MXFP4` with TP8, AITER, FP8 KV cache, and EAGLE crashed during a heterogeneous long-horizon agent workload with 32 concurrent clients.

The server ran normally for about 24 minutes under this load, then two ranks reported GPU memory access faults and `scheduler_4` exited with `-6`. The watchdog shut down the whole server. Docker reported `OOMKilled=false`; the coordinator ultimately exited `0` after cleanup. Starting the same container again, without changing the checkpoint or argv, restored `/health`, `/v1/models`, and tool calling.

This currently looks more like a SGLang/AITER/EAGLE high-concurrency runtime fault than checkpoint corruption. It is very similar to #23784, which also reports MI355X, EAGLE, high concurrency, untuned large BF16 GEMM shapes, and the same KFD fault signature.

The known DSA MQA-logits 2 GiB failure is not the observed signature here. An equivalent of #36960 was active at runtime:

```text
MQA_CAP=patch
[patch] applied _MQA_LOGITS_MAX_BYTES_ROCM = 2**31 - 1
```

That known failure aborts in LLVM with an `iota_range Begin <= End` assertion. This incident instead produced a GPU runtime memory access fault.

### Reproduction

This is the full reproducing workload, not yet a minimal standalone script.

Server flags:

```bash
python3 -m sglang.launch_server \
  --model-path /model \
  --served-model-name glm-5.3 \
  --trust-remote-code \
  --tp 8 \
  --kv-cache-dtype fp8_e4m3 \
  --speculative-algorithm EAGLE \
  --speculative-num-steps 3 \
  --speculative-eagle-topk 1 \
  --speculative-num-draft-tokens 4 \
  --page-size 64 \
  --dsa-prefill-backend tilelang \
  --dsa-decode-backend tilelang \
  --reasoning-parser glm45 \
  --tool-call-parser glm47 \
  --chat-template /model/chat_template.jinja \
  --enable-cache-report \
  --chunked-prefill-size 131072 \
  --mem-fraction-static 0.80 \
  --cuda-graph-max-bs 64 \
  --max-running-requests 64 \
  --watchdog-timeout 1200 \
  --host 0.0.0.0 \
  --port 30000
```

Relevant environment:

```text
SGLANG_USE_AITER=1
AITER_COMMIT=c16d44b93a528b2a4bfd6d8d3409116d465872a9
AITER_USE_SYSTEM_TRITON=1
SGLANG_OPT_USE_TOPK_V2=false
SGLANG_ROCM_FUSED_DECODE_MLA=1
SGLANG_MOE_PADDING=1
SGLANG_ROCM_DISABLE_LINEARQUANT=0
ROCM_QUICK_REDUCE_QUANTIZATION=INT8
HSA_ENABLE_IPC_MODE_LEGACY=1
HSA_NO_SCRATCH_RECLAIM=1
HIP_FORCE_DEV_KERNARG=1
MQA_CAP=patch
```

Client workload:

- 32 concurrent Claude Code 2.1.207 agent sessions.
- Mixed multi-turn coding workloads with growing and heterogeneous contexts.
- Immediately before the fault, the scheduler reported 24-29 running requests, no queued requests, and token usage around 0.72-0.83 of a ~3.68M-token pool.
- Fixed-shape serving tests at concurrency 64 with 8K input / 1K output had completed previously; the varied long-context workload is what exposed the fault.

Last scheduler observations included:

```text
[2026-09-02 07:49:58 TP0] Prefill batch, #new-seq: 1, #new-token: 256,
#cached-token: 151360, token usage: 0.72, #running-req: 24,
#queue-req: 0, cuda graph: False

[aiter] shape is M:84333, N:2624, K:6144 dtype='torch.bfloat16' ...
not found tuned config ... will use default config! using torch solution:0
[aiter] shape is M:84333, N:512, K:6144 dtype='torch.bfloat16' ...
not found tuned config ... will use default config! using torch solution:0
[aiter] shape is M:84333, N:6144, K:256 dtype='torch.bfloat16' ...
not found tuned config ... will use default config! using torch solution:0

Memory access fault by GPU node-8 (Agent handle: 0xcbf2ae0)
on address 0x7fb5cd056000. Reason: Unknown.
Memory access fault by GPU node-6 (Agent handle: 0xc9023d0)
on address 0x7fc870f50000. Reason: Unknown.

[2026-09-02 07:50:00] Subprocess scheduler_4 (pid=776)
crashed with exit code -6. Triggering SIGQUIT for cleanup...
```

The first investigation pass did not preserve a useful GPU core dump; the handler exited with status 1.

### Environment

`python3 -m sglang.check_env`:

```text
Python: 3.12.3
ROCM available: True
GPU 0,1,2,3,4,5,6,7: AMD Instinct MI355X
GPU Compute Capability: 9.5
ROCM_HOME: /opt/rocm-7.2.4
HIPCC: HIP version: 7.2.53211-97f5574fe2
ROCM Driver Version: 6.16.6
PyTorch: 2.11.0+rocm7.2
sglang: 0.5.18.dev20260829+g4d53767b09
sglang-kernel: 0.4.6.post1
triton: 3.7.0+amd.rocm7.2.0.git89002410
transformers: 5.12.1
numpy: 2.3.5
xgrammar: 0.2.1
anthropic: 1.2.0
AMD topology: full XGMI between all 8 GPUs
ulimit soft: 1048576
```

Additional pins:

```text
Container image: rocm/sgl-dev:v0.5.18-rocm724-mi35x-20260829
Image ID: sha256:95e86f0d9ae3693bf440e0e8df14225040fa5001097703d4d867891a53360a5d
SGLang source checkout visible in image: cdbfe90b4a6c728e03e6520862d792501b3a97bb
AITER: c16d44b93a528b2a4bfd6d8d3409116d465872a9
AMD Quark checkpoint format: 0.12.post1+1b229f7
Model index geometry: index_n_heads=32, index_head_dim=128, index_topk=2048
```

The deployed SGLang revision is behind current `main`; this has not yet been reproduced on current `main`.

### Scope and current hypothesis

The model provider's validated configuration is substantially narrower: TP4/EP4, `max-running-requests=2`, `cuda-graph-max-bs=2`, and a different stock SGLang image digest. Our TP8/high-concurrency configuration is outside that validated envelope, so this report does not claim a defective checkpoint.

Current hypotheses, in descending order:

1. The high-concurrency EAGLE draft path described in #23784.
2. A graph-backed AITER kernel selected only at the larger live batch/context shapes.
3. An untuned large-shape fallback around the `M=84333` prefill.

Planned follow-up matrix:

- EAGLE on vs off.
- CUDA graph max batch 2 vs 16/64, then eager mode if needed.
- Client concurrency 8 vs 16 vs 32.
- AITER MoE/default path ablation.
- A smaller shape-driven reproducer and a usable GPU core dump.

Could maintainers advise whether #23784 had a concrete fix that is expected in current `main`, and which EAGLE/AITER boundary would be most useful to instrument first? I will update this issue with the ablation results when we return to the investigation.

Related:

- #23784
- #36960
- ROCm/aiter#5114
- ROCm/aiter#3737
